testsuite/init_module.c

```diff
@@ -234,17 +234,24 @@ long init_module(void *mem, unsigned long len, const char *args)
 
 	init_retcodes();
 
-	elf = kmod_elf_new(mem, len);
-	if (elf == NULL)
-		return 0;
+	if (mem == NULL) {
+		errno = EFAULT;
+		return -1;
+	}
+
+	err = kmod_elf_new(mem, len, &elf);
+	if (err < 0) {
+		errno = -err;
+		return -1;
+	}
 
 	err = kmod_elf_get_section(elf, ".gnu.linkonce.this_module", &off, &bufsize);
 	buf = (const char *)kmod_elf_get_memory(elf) + off;
 	kmod_elf_unref(elf);
 
 	/* We couldn't parse the ELF file. Just exit as if it was successful */
 	if (err < 0)
-		return 0;
+		return -1;
```

Comment:
If we really want to return an error now (which I don't mind), the comment has to be adjusted or removed.